homeassistant/components/modbus/binary_sensor.py

```diff
@@ -61,10 +111,46 @@ async def async_update(self, now: datetime | None = None) -> None:
                 return
             self._lazy_errors = self._lazy_error_count
             self._attr_available = False
-            self.async_write_ha_state()
-            return
+            self._result_bits = None
+        else:
+            self._lazy_errors = self._lazy_error_count
+            self._attr_is_on = result.bits[0] & 1
+            self._attr_available = True
+            self._result_bits = result.bits
 
-        self._lazy_errors = self._lazy_error_count
-        self._attr_is_on = result.bits[0] & 1
-        self._attr_available = True
         self.async_write_ha_state()
+        if self._coordinator:
+            await self._coordinator.async_refresh()
```

Comment:
Don't do refresh, you can just set data instead.
```suggestion
            self._coordinator.async_set_updated_data(self._result_bits)
```